plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/ScalaLanguageModelBuilder.java

```diff
@@ -6,9 +6,21 @@
 import java.io.File;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 
+import com.microsoft.java.bs.gradle.model.GradleModuleDependency;
+import com.microsoft.java.bs.gradle.model.ScalaExtension;
+import com.microsoft.java.bs.gradle.model.SupportedLanguage;
+import com.microsoft.java.bs.gradle.model.impl.DefaultScalaExtension;
 import org.gradle.api.Project;
 import org.gradle.api.file.SourceDirectorySet;
+import org.gradle.api.internal.tasks.scala.MinimalScalaCompileOptions;
```

Comment:
Do you think it's necessary/possible to have some kind of fallback in case some breaking changes happens to those internal apis?

Reply:
I'd say that's more of a question across the whole codebase.  If you do a regex search on `\.gradle.*\.internal` you can see that internal apis are used in 5 different files.
From my experience with Gradle since around version 5, any changes to the Java/Scala APIs have been clearly deprecated for a few versions before being removed.
I guess part of it is how quickly you could respond to an API change by putting out a new version and having that picked up by VSCode.  There should be plenty of time with the deprecation warnings.
I'd like to see this project published to Maven so that other BSP clients (e.g. [Metals](https://github.com/scalameta/metals)) could easily install it and pick up any fixes.  AFAIK at the moment the only way to install this BSP server is to install the VSCode extension but lots of people use other UIs.  Could you look to publishing it on Maven?  I could raise an issue but it's not something I could submit a PR for.  There are lots of microsoft java projects up on Maven so maybe one of your colleagues can help.

Reply:
> I guess part of it is how quickly you could respond to an API change by putting out a new version and having that picked up by VSCode. There should be plenty of time with the deprecation warnings.

Makes sense!

> Could you look to publishing it on Maven?

Filed an issue: https://github.com/microsoft/build-server-for-gradle/issues/129, I'll discuss with my teammates about this.